### PR TITLE
A0-1175: Full genesis init for pallet elections

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

This PR introduces two changes:
 - we enable setting non-reserved seats for genesis build as well
 - we initialize all the (reasonable) storage items in genesis build 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have added tests
- I have bumped `spec_version`

### CODE REVIEW

File: pallets/elections/src/tests.rs [hunk 1]
```diff
@@ -14,54 +21,86 @@ fn support(total: Balance, voters: Vec<(AccountId, Balance)>) -> Support<Account
     Support { total, voters }
 }
 
+#[test]
+fn storage_is_initialized_already_in_genesis() {
+    static RESERVED: [AccountId; 3] = [1, 2, 3];
+    static NON_RESERVED: [AccountId; 2] = [4, 5];
+    static COMMITTEE_SEATS: CommitteeSeats = CommitteeSeats {
```
Comment (kostekIV) [inline pallets/elections/src/tests.rs:28]: use const instead of static

File: pallets/elections/src/lib.rs [hunk 2]
```diff
@@ -48,7 +51,8 @@ impl<AccountId> Default for EraValidators<AccountId> {
     }
 }
 
-#[derive(Decode, Encode, TypeInfo, Debug, Clone, Copy, PartialEq, Eq, Default)]
+#[derive(Decode, Encode, TypeInfo, Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
```
Comment (kostekIV) [inline pallets/elections/src/lib.rs:55]: we can move this to primitives to avoid adding `serde` dependency to the pallet

**Review by kostekIV — APPROVED** [review]

File: pallets/elections/src/lib.rs (hunk 2 above)
Comment (pmikolajczyk41) [reply pallets/elections/src/lib.rs:55]: the other alternative is to take just `reserved_seats: u32` and `non_reserved_seats: u32` in `GenesisConfig`; this has such an advantage that `CommitteeSeats` would stay in pallet where it belongs; wdyt?

**Review by obrok — APPROVED** [review]
LGTM

File: pallets/elections/src/lib.rs (hunk 2 above)
Comment (kostekIV) [reply pallets/elections/src/lib.rs:55]: I would move it to primitives anyway, an additional reason is that this struct is used in other places as well (e2e tests, chain_spec.rs etc)

Comment (pmikolajczyk41) [reply pallets/elections/src/lib.rs]: done

**Pull request merged**